src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,459 @@
+
+        if doc_stride is not None:
+            # TODO implement
+            # doc_stride = min(max_seq_len // 2, 128)
+            raise ValueError("Unsupported: striding inputs")
+
+        image = None
+        image_features = {}
+        if "image" in input:
+            image = load_image(input["image"])
+            if self.feature_extractor is not None:
+                image_features.update(self.feature_extractor(images=image, return_tensors=self.framework))
+            elif self.model_type == ModelType.Donut:
+                raise ValueError("If you are using a VisionEncoderDecoderModel, you must provide a feature extractor")
+
+        words, boxes = None, None
+        if not self.model_type == ModelType.Donut:
+            if "word_boxes" in input:
+                words = [x[0] for x in input["word_boxes"]]
+                boxes = [x[1] for x in input["word_boxes"]]
+            elif "words" in image_features and "boxes" in image_features:
+                words = image_features.pop("words")[0]
+                boxes = image_features.pop("boxes")[0]
+            elif image is not None:
+                if not TESSERACT_LOADED:
+                    raise ValueError(
+                        "If you provide an image without word_boxes, then the pipeline will run OCR using Tesseract,"
+                        " but pytesseract is not available"
+                    )
+                if TESSERACT_LOADED:
+                    words, boxes = apply_tesseract(image, lang=lang, tesseract_config=tesseract_config)
+            else:
+                raise ValueError(
+                    "You must provide an image or word_boxes. If you provide an image, the pipeline will automatically"
+                    " run OCR to derive words and boxes"
+                )
+
+        if self.tokenizer.padding_side != "right":
+            raise ValueError(
+                "Document question answering only supports tokenizers whose padding side is 'right', not"
+                f" {self.tokenizer.padding_side}"
+            )
+
+        if self.model_type == ModelType.Donut:
+            task_prompt = f'<s_docvqa><s_question>{input["question"]}</s_question><s_answer>'
+            # Adapted from https://huggingface.co/spaces/nielsr/donut-docvqa/blob/main/app.py
+            encoding = {
+                "inputs": image_features["pixel_values"],
+                "decoder_input_ids": self.tokenizer(
+                    task_prompt, add_special_tokens=False, return_tensors=self.framework
+                ).input_ids,
+                "max_length": self.model.decoder.config.max_position_embeddings,
+                "early_stopping": True,
+                "pad_token_id": self.tokenizer.pad_token_id,
+                "eos_token_id": self.tokenizer.eos_token_id,
+                "use_cache": True,
+                "num_beams": 1,
+                "bad_words_ids": [[self.tokenizer.unk_token_id]],
+                "return_dict_in_generate": True,
```

Comment:
I assume that parameters related to generation are all configuration in the `text-generation` pipeline, @Narsil?

Reply:
could you give me some guidance on whether I should change anything? E.g. should I make some of these parameters of the pipeline?

Reply:
Yes, remove everything you can:

```python
encoding = {"inputs": image_feature["pixel_values"], "decoder_input_ids": self.tokenizer(..).input_ids}
```
should be enough in most cases I think (`generate` should recover everything that's missing on its own.)